readthedocs/projects/tasks/utils.py

```diff
@@ -154,6 +160,81 @@ def send_external_build_status(version_type, build_pk, commit, status):
         send_build_status.delay(build_pk, commit, status)
 
 
+class DeprecatedConfigFileSiteNotification(SiteNotification):
+
+    failure_message = (
+        "Your project '{{ object.slug }}' doesn't have a "
+        '<a href="https://docs.readthedocs.io/en/stable/config-file/v2.html">.readthedocs.yaml</a> '
+        "configuration file. "
+        "This feature is <strong>deprecated and will be removed soon</strong>. "
```

Comment:
I guess I wouldn't describe this as a feature to users. I'd probably stick with describing the outcome to the user, instead of describing the change we're making.

```suggestion
        "Configuration files will <strong>soon be required</strong> by projects, and will no longer be optional. "
```

Maybe "optional" is redundant, but it describes the current state to the user.